README.md

```diff
@@ -6,6 +8,24 @@ network connectivity. Connectivity is checked between the server part and multit
 of agents. The agents then make periodical reports to the server and presence of
 reported data indicates aliveness of the link.
 
+How it works
+============
+
+Network checker is an Kubernetes application main purpose of which is checking of connectivity
+between the cluster's nodes. Network checker consists of two parts: server (this repository)
+and agent ([developed here](https://github.com/Mirantis/k8s-netchecker-agent)). Agents are deployed on
+every K8S node using [Daemonset mechanism](https://kubernetes.io/docs/admin/daemons/)
+(to ensure auto-management of the pods). Agents come in two flavors - and there exists two daemonsets
+for each of them. The difference between them is that "Agent-hostnet" is tapped into host network namespace
```

Comment:
... two daemonsets for each of them.

Probably, "... two daemonsets for each agent." is better as it is not clear what is "them" here.